File: android/sharedui/components/AppIcon.kt
package com.zhiqihuayun.sharedui.components

import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.ColorFilter
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.zhiqihuayun.foundation.design.AppColor
import com.zhiqihuayun.sharedui.icons.SfApproxIcons

/**
 * 中台统一图标名（双端共用，iOS 映射 SF Symbol，Android 映射 SfApproxIcons）。
 */
enum class AppIconName(val label: String) {
    List("列表"),
    Chart("图表"),
    Plus("加号"),
    Safari("浏览器"),
    Person("人物"),
    ArrowDown("下拉三角"),
    Smartphone("手机"),
    Mail("邮箱");

    companion object {
        /** 对应的 iOS SF Symbol 系统名（供跨端文档参考）。 */
        fun sfSymbolName(self: AppIconName): String = when (self) {
            List -> "list.bullet.rectangle"
            Chart -> "chart.xyaxis.line"
            Plus -> "plus.circle.fill"
            Safari -> "safari"
            Person -> "person"
            ArrowDown -> "arrowtriangle.down.fill"
            Smartphone -> "smartphone"
            Mail -> "envelope"
        }
    }
}

/**
 * 中台基础图标组件（Compose）。
 *
 * 封装 SfApproxIcons 矢量图标，统一双端图标使用方式。
 * Android 用 SfApproxIcons 手绘矢量近似 SF Symbols，iOS 直接用系统 SF Symbols。
 *
 * @param name 图标名（双端统一枚举）
 * @param modifier 布局修饰符
 * @param size 尺寸（dp），默认 24
 * @param tint 着色，默认 textPrimary
 */
@Composable
fun AppIcon(
    name: AppIconName,
    modifier: Modifier = Modifier,
    size: Dp = 24.dp,
    tint: Color = AppColor.textPrimary
) {
    val vector: ImageVector = when (name) {
        AppIconName.List -> SfApproxIcons.ListBulletRectangle
        AppIconName.Chart -> SfApproxIcons.ChartXyAxisLine
        AppIconName.Plus -> SfApproxIcons.PlusCircleFill
        AppIconName.Safari -> SfApproxIcons.Safari
        AppIconName.Person -> SfApproxIcons.Person
        AppIconName.ArrowDown -> SfApproxIcons.ArrowTriangleDownFill
        AppIconName.Smartphone -> SfApproxIcons.Smartphone
        AppIconName.Mail -> SfApproxIcons.Mail
    }
    Image(
        imageVector = vector,
        contentDescription = name.label,
        modifier = modifier.size(size),
        colorFilter = ColorFilter.tint(tint),
        contentScale = ContentScale.Fit
    )
}

File: ios/SharedUI/Components/AppIcon.swift
/// 中台基础图标组件（UIKit 版，对齐 Android AppIcon）。
///
/// 封装 SF Symbols，统一双端图标使用方式。
/// iOS 直接用系统 SF Symbols（零资源开销），Android 用 SfApproxIcons 矢量近似。
///
/// 用法：
/// ```swift
/// let icon = AppIcon.make(.list, size: 24, color: AppColor.primary)
/// view.addSubview(icon)
/// ```

import UIKit

/// 中台统一图标名（双端共用，iOS 映射 SF Symbol，Android 映射 SfApproxIcons）。
enum AppIconName: String, CaseIterable {
    case list
    case chart
    case plus
    case safari
    case person
    case arrowDown
    case smartphone
    case mail

    /// 对应的 SF Symbol 系统名。
    var sfSymbol: String {
        switch self {
        case .list: return "list.bullet.rectangle"
        case .chart: return "chart.xyaxis.line"
        case .plus: return "plus.circle.fill"
        case .safari: return "safari"
        case .person: return "person"
        case .arrowDown: return "arrowtriangle.down.fill"
        case .smartphone: return "iphone.gen3"
        case .mail: return "envelope"
        }
    }

    /// 语义标签（无障碍 / Demo 展示用）。
    var label: String {
        switch self {
        case .list: return "列表"
        case .chart: return "图表"
        case .plus: return "加号"
        case .safari: return "浏览器"
        case .person: return "人物"
        case .arrowDown: return "下拉三角"
        case .smartphone: return "手机"
        case .mail: return "邮箱"
        }
    }
}

/// 中台基础图标组件（UIKit）。
final class AppIcon: UIImageView {
    /// 图标名。
    private(set) var iconName: AppIconName = .list

    /// 默认尺寸。
    static let defaultSize: CGFloat = 24

    // MARK: - 工厂方法

    /// 创建图标。
    /// - Parameters:
    ///   - name: 图标名（双端统一枚举）。
    ///   - size: 尺寸（pt），默认 24。
    ///   - color: 着色（tintColor），默认 textPrimary。
    /// - Returns: 配置好的 AppIcon 实例，已约束宽高。
    static func make(
        _ name: AppIconName,
        size: CGFloat = defaultSize,
        color: UIColor = AppColor.textPrimary
    ) -> AppIcon {
        let icon = AppIcon()
        icon.iconName = name
        // 强制 SF Symbol 单色（monochrome）轮廓渲染，与 Android SfApproxIcons 矢量着色语义一致。
        // 否则「iphone.gen3」等多色符号会保留内部屏幕渐变/Home Indicator 颜色，
        // 忽略 tintColor，导致双端视觉不一致（iOS 有内色 vs Android 纯白/纯色）。
        let config = UIImage.SymbolConfiguration.preferringMonochrome()
        icon.image = UIImage(systemName: name.sfSymbol, withConfiguration: config)
        icon.tintColor = color
        icon.contentMode = .scaleAspectFit
        icon.translatesAutoresizingMaskIntoConstraints = false
        icon.widthAnchor.constraint(equalToConstant: size).isActive = true
        icon.heightAnchor.constraint(equalToConstant: size).isActive = true
        return icon
    }

    /// 更新着色。
    func setColor(_ color: UIColor) {
        tintColor = color
    }
}
